src/many-cli/src/main.rs

```diff
@@ -159,10 +182,41 @@ fn main() {
             }
         }
         SubCommand::Message(o) => {
-            // If `pem` is not provided, use anonymous and don't sign.
-            let key = o.pem.map_or_else(CoseKeyIdentity::anonymous, |p| {
-                CoseKeyIdentity::from_pem(&std::fs::read_to_string(&p).unwrap()).unwrap()
-            });
+            let key = if o.module.is_some() && o.slot.is_some() && o.keyid.is_some() {
+                trace!("Getting user PIN");
+                let pin = rpassword::prompt_password("Please enter the HSM user PIN: ").unwrap();
+                let keyid = hex::decode(o.keyid.unwrap()).expect("Failed to decode keyid to hex");
+
+                {
+                    let mut hsm = HSM_INSTANCE.lock().unwrap();
```

Comment:
Can you make it a function of `hsm`? Like

```rust
let mut hsm: MutexGuard<HSM> = HSM::get_intance();
```

Reply:
I wrote it so it returns a `Result` since this thing can panic.